pkg/apis/ceph.rook.io/v1/validate_pools.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"github.com/coreos/pkg/capnslog"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	webhookName = "rook-ceph-webhook"
+	logger      = capnslog.NewPackageLogger("github.com/rook/rook", webhookName)
+)
+
+var _ webhook.Validator = &CephBlockPool{}
+
+func (p *CephBlockPool) ValidateCreate() error {
+	logger.Infof("validate create cephblockpool %v", p)
+
+	err := ValidatePoolSpecs(p.Spec)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidatePoolSpecs(ps PoolSpec) error {
+	// Checks if either ErasureCoded or Replicated fields are set
+	if ps.ErasureCoded.CodingChunks <= 0 && ps.ErasureCoded.DataChunks <= 0 && ps.Replicated.TargetSizeRatio <= 0 && ps.Replicated.Size <= 0 {
```

Comment:
We really should have unit tests for all of these validations. It should be fairly simple to create them, so would be great to have them with this PR.

Reply:
Working on it

Reply:
Updated with unit tests. please review